tests/validation/tests/v3_api/test_hpa.py

```diff
@@ -0,0 +1,499 @@
+import pytest
+from rancher import ApiError
+from .common import *  # NOQA
+
+namespace = {"p_client": None, "ns": None, "cluster": None, "project": None}
+
+
+def test_create_hpa():
+    p_client = namespace["p_client"]
+    ns = namespace["ns"]
+    hpa = create_hpa(p_client, ns)
+    p_client.delete(hpa)
+
+
+def test_edit_hpa():
+    p_client = namespace["p_client"]
+    ns = namespace["ns"]
+    hpa = edit_hpa(p_client, ns)
+    p_client.delete(hpa)
+
+
+def test_delete_hpa():
+    p_client = namespace["p_client"]
+    ns = namespace["ns"]
+    hpa = create_hpa(p_client, ns)
+    delete_hpa(p_client, hpa, ns)
+
+
+# Cluster member is not added in the role list below as this role does not
+# have access to the rbac created project and he has to create his own project
+
+rbac_role_list = [
+    (CLUSTER_OWNER),
+    (PROJECT_OWNER),
+    (PROJECT_MEMBER),
+]
+
+
+@if_test_rbac
+@pytest.mark.parametrize("role", rbac_role_list)
+def test_rbac_hpa_create(role):
+    user_token = rbac_get_user_token_by_role(role)
+    project = rbac_get_project()
+    ns = rbac_get_namespace()
+    p_client = get_project_client_for_token(project, user_token)
+    create_hpa(p_client, ns)
+
+
+@if_test_rbac
+@pytest.mark.parametrize("role", rbac_role_list)
+def test_rbac_hpa_edit(role):
+    user_token = rbac_get_user_token_by_role(role)
+    project = rbac_get_project()
+    ns = rbac_get_namespace()
+    p_client = get_project_client_for_token(project, user_token)
+    edit_hpa(p_client, ns)
+
+
+@if_test_rbac
+@pytest.mark.parametrize("role", rbac_role_list)
+def test_rbac_hpa_delete(role):
+    user_token = rbac_get_user_token_by_role(role)
+    project = rbac_get_project()
+    ns = rbac_get_namespace()
+    p_client = get_project_client_for_token(project, user_token)
+    hpa = create_hpa(p_client, ns)
+    delete_hpa(p_client, hpa, ns)
+
+
+@if_test_rbac
+@pytest.mark.parametrize("role", rbac_role_list)
+def test_rbac_hpa_list(remove_resource, role):
+    user_token = rbac_get_user_token_by_role(role)
+    project = rbac_get_project()
+    ns = rbac_get_namespace()
+    user_client = get_project_client_for_token(project, user_token)
+    hpa = create_hpa(user_client, ns)
+    hpaname = hpa["name"]
+    hpadict = user_client.list_horizontalPodAutoscaler(name=hpaname)
+    print(hpadict)
+    assert hpa.type == "horizontalPodAutoscaler"
```

Comment:
it is better to check the object returned by the list method, not the one returned by the create method.

Reply:
Done